### PR TITLE
sql/(parse,expression): implement unary minus

### PR DESCRIPTION
**Pull request by erizocosmico**
Closes https://github.com/src-d/go-mysql-server/issues/444

### CODE REVIEW

File: sql/expression/arithmetic.go [hunk 1]
```diff
@@ -407,3 +408,81 @@ func mod(lval, rval interface{}) (interface{}, error) {
...
+	}
+
+	if !sql.IsNumber(e.Child.Type()) {
+		child, err = sql.Float64.Convert(child)
+		if err != nil {
+			child = 0.0
+		}
+	}
+
+	switch n := child.(type) {
```
Comment (kuba--) [inline sql/expression/arithmetic.go]: don't we need a case for regular `int`?

Comment (kuba--) [reply sql/expression/arithmetic.go]: I think for:
```
var child interface{} = 10
```
it will go to the default.

**Comment by ajnavarro** [comment]
@erizocosmico can you rebase please?

**Comment by erizocosmico** [comment]
Done

**Comment by ajnavarro** [comment]
@erizocosmico could you add a test for:

```
case *sqlparser.UnaryExpr:
	return unaryExprToExpression(v)
```

**Comment by erizocosmico** [comment]
Done

**Pull request merged**